src/main/java/org/datadog/jmxfetch/reporter/Reporter.java

```diff
@@ -119,7 +119,15 @@ public void sendMetrics(List<Metric> metrics, String instanceName, boolean canon
                 }
                 sendMetricPoint(metricType, metricName, delta, tags);
 
-            } else { // The metric should be 'counter'
+            } else {
+                // `counter` and `rate` are equivalent and both accepted as valid.
+                // We continue accepting other cases for backward compatibility.
+                if (!"counter".equals(metricType) && !"rate".equals(metricType)) {
+                    log.warn("Invalid metric type " + metricType + " for metric " + metricName
```

Comment:
I think the warning level is way too verbose, if someone triggers this code path this will be all over the logs.

Reply:
That's right, the upside is that it will motivate to fix the issue.

If we use info, the quantity of output will be same and less visible.

If we use debug, people won't see it and/or ignore it.

Reply:
@truthbk An alternative is the have an boolean attribute like `this.isInvalidMetricTypeWarningLogged` to log only once:

```java
                if (!metricType.equals("counter")
                        && !metricType.equals("rate")
                        && !this.isInvalidMetricTypeWarningLogged) {
                    log.warn("Invalid metric type " + metricType + " for metric " + metricName
                            + ". The metric will be processed as rate"
                            + " (this is a DEPRECATED behaviour, use a valid type instead).");
                    this.isInvalidMetricTypeWarningLogged = true;
                }
```

WDYT?

Reply:
>  it will motivate to fix the issue
Yeah, that's for sure! 😁 

But some customers ingest agent logs, and those are billed, so I'm pretty sure we shouldn't log at the warn level. As to adding a flag like that, it could work. Printing it just once might not be enough, especially with the log rotation. Maybe lowering it to `debug` is enough, or if you want to make it stand out in `warn`, you could maybe use an integer and set a desired frequency to skip, and print every N iterations or something - it would add some overhead, but only for the bad types so maybe not the end of the world.

Reply:
Let's use debug for now then :)

Changed.